charybdis/src/services/dump.service.ts

```diff
@@ -0,0 +1,257 @@
+import { prisma as localPrisma } from "../local-prisma/prisma";
+import * as fs from "fs";
+import * as path from "path";
+import { v4 as uuidV4 } from "uuid";
+import { LocalRun } from "../types/local.types";
+import { CsvRunRow } from "../types/csv.types";
+import { DOWNLOADS_PATH } from "../storage-paths";
+import {
+  CouldNotConnectToLocalDB,
+  DataTypeDumpFailed,
+  RunDumpFailed,
+  DataDumpFailed,
+} from "../errors/dump.errors";
+import { appendToCsv, prependToCsv } from "../utils/csv.utils";
+import {
+  createFile,
+  createFolder,
+  createMeaningfulFileName,
+} from "../utils/filesystem.utils";
+import { FailedWriteAuditLog } from "../errors/audit.errors";
+
+async function checkDbConnection() {
+  try {
+    await localPrisma.$connect();
+  } catch (error) {
+    throw new CouldNotConnectToLocalDB();
+  }
+}
+
+async function initializeDumpFileStructure(): Promise<{
+  dumpFolderPath: string;
+  auditLogCsv: string;
+  dataTypesCsv: string;
+  runsCsv: string;
+  dataFolder: string;
+}> {
+  console.log("Acquiring dump file paths...");
+  const currentDumpName = createMeaningfulFileName("dump", new Date());
+  const dumpFolderPath = `${DOWNLOADS_PATH}/${currentDumpName}`;
+  const auditLogCsv = `${DOWNLOADS_PATH}/audit_log.csv`;
+  await createFolder(DOWNLOADS_PATH);
+  await createFolder(dumpFolderPath);
+  await createFile(auditLogCsv);
+  const dataTypesCsv = `${dumpFolderPath}/data_type.csv`;
+  await createFile(dataTypesCsv);
+  const runsCsv = `${dumpFolderPath}/run.csv`;
+  await createFile(runsCsv);
+  const dataFolder = `${dumpFolderPath}/data`;
+  await createFolder(dataFolder);
+  return { dumpFolderPath, auditLogCsv, dataTypesCsv, runsCsv, dataFolder };
+}
+
+export async function dumpLocalDb(): Promise<void> {
+  console.log("Checking database connection...");
+  // check that we can actually connect to the database
+  await checkDbConnection(); // throws if prisma cannot connect to local
+  const { dumpFolderPath, auditLogCsv, dataTypesCsv, runsCsv, dataFolder } =
+    await initializeDumpFileStructure();
+  console.log("Starting dump process...");
+  try {
+    try {
```

Comment:
just wrap the functions themselves in try catches rather than the function calls

Reply:
this function should have one try catch.